conda/core/package_cache_data.py

```diff
@@ -275,17 +260,15 @@ def _check_writable(self):
         return i_wri
 
     @staticmethod
-    def _clean_tarball_path_and_get_checksums(tarball_path, sha256sum=None, md5sum=None):
+    def _clean_tarball_path_and_get_md5sum(tarball_path, md5sum=None):
```

Comment:
It looks like you are reverting a lot of the work I did to prefer sha256 over md5 where possible.  Is that intentional?  What's your reasoning on that?

Reply:
The purpose of using md5 in all of these places is only for quick identification, and to have reasonable assurance that we have the "right" artifact, in the context of distinguishing if the package `flask-1.0.2-py_0.tar.bz2` in the package cache came from `free`, `main`, `conda-forge`, or some other channel.  We know pretty close to what it should be by the file name, but because we don't include channel information with the bare tarballs in the package cache, we don't know what channel they come from.  It's also a quick check to ensure that a package hasn't been updated in-place in the remote channel (and therefore needs to be downloaded again).

Matching the md5 to the expected md5 from repodata gives us the assurance that an artifact that has the right name in a package cache is actually the artifact that we want.  To be clear: **Using md5 here is not a security measure.**  Just as we use md5 as a checksum to ensure downloads have not been _inadvertently_ corrupted, we use md5 here for a fast check to make sure the artifact is as we expect (or not).

Neither md5 nor sha256 in this context is sufficient as a security measure to guard against malicious attacks.  The purpose of using md5 here is ONLY as a quick checksum.  And in these quick checksum cases we should prefer md5 to sha256 since the former is indeed quite a bit faster.

When we add cryptographic signatures, we will be making specific and more targeted use of sha256 + size, in combination.

Reply:
Thanks.  That makes sense.  Is it worth keeping at least for the file downloads, though?

Reply:
Yeah, I don't see any problem with that.  And while we're at it, we might as well add in size to that function too and get rid of one of those TODOs.  Pushing another commit soon with the change.